public/app/plugins/datasource/grafana-azure-monitor-datasource/module.ts

```diff
@@ -16,16 +16,22 @@ getAppEvents().subscribe<DashboardLoadedEvent<AzureMonitorQuery>>(
   DashboardLoadedEvent,
   ({ payload: { dashboardId, orgId, userId, grafanaVersion, queries } }) => {
     const azureQueries = queries[pluginJson.id];
+    let stats: { [key: string]: number } = {};
+    azureQueries.forEach((query) => {
+      const statName =
+        (query.queryType?.toLowerCase().replaceAll(' ', '_') ?? 'unknown') +
+        '_queries_' +
+        (query.hide ? 'hidden' : 'executed');
+      stats[statName] = ~~stats[statName] + 1;
```

Comment:
This is a neat way, and will scale if we add more query types, but that means the event structure will change a few times for RS/BQ (first event probably won't have all types of query hidden and executed, so columns will be added by subsequent events triggered).

Maybe it's best I initialise all 6 props (Monitor, Logs, ARG for hidden and executed) to 0 to avoid the schema changes?

Reply:
Yeah I think declaring the props will be easier to understand.

Reply:
Ditto.